error: ‘bool’ cannot be defined via ‘typedef’

I meant to report this a while back, but I'm trying to update the Arch AUR package for TileDB, however, it fails with:

`error: ‘bool’ cannot be defined via ‘typedef’`

```bash
/home/user/Downloads/AUR/tiledb/src/TileDB-2.28.0/external/blosc/src/shuffle.c:30:15: error: ‘bool’ cannot be defined via ‘typedef’
   30 | typedef _Bool bool;
      |               ^~~~
/home/user/Downloads/AUR/tiledb/src/TileDB-2.28.0/external/blosc/src/shuffle.c:30:15: note: ‘bool’ is a keyword with ‘-std=c23’ onwards
```

The issue is due to GCC-15, which defaults to `‘-std=c23’`.

I'm currently only testing against v2.28.0, but I can see that the same failure will happen on v2.28.1 as no changes to blosc between those two versions.

My current workaround is via this patch, but it might be better to resolve the issue rather than me kicking it down the road and ignoring it (as well as the issue of me forcing a standard).

```patch
diff --git a/external/blosc/CMakeLists.txt b/external/blosc/CMakeLists.txt
index d9bbb3599..97e799bd1 100644
--- a/external/blosc/CMakeLists.txt
+++ b/external/blosc/CMakeLists.txt
@@ -73,6 +73,7 @@ endif()
 # The blosc files are external and we ignore their warnings.
 # -w is common to gcc and MSVC
 set(TILEDB_BLOSC_COMPILE_OPTIONS "-w")
+string(APPEND TILEDB_BLOSC_COMPILE_OPTIONS ";-std=c17")
 # If we found pthreads, activate it by defining PTHREAD_AVAILABLE.
 if (CMAKE_USE_PTHREADS_INIT AND NOT MSVC)
   string(APPEND TILEDB_BLOSC_COMPILE_OPTIONS ";-DPTHREAD_AVAILABLE")
```

Any thoughts? Is there a better alternative I should use other than the patch above, which forces a C standard in the CMake compile options?

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -107,6 +107,12 @@ else()
   set(CMAKE_CXX_EXTENSIONS OFF)
 endif()
 
+# Set C99 as default required standard for all C targets.
+# Some vendored source files are written in C. We should remove
+# this when we switch to vcpkg.
+set(CMAKE_C_STANDARD 99)
+set(CMAKE_C_STANDARD_REQUIRED ON)
+
 if (TILEDB_CCACHE)
   include(FindCcache)
   set(CMAKE_CXX_COMPILER_LAUNCHER ${CCACHE})

